sdk/trace/build.gradle.kts

```diff
@@ -69,10 +76,9 @@ tasks {
         }
     }
 
-    jar {
-        inputs.files(project(":sdk:trace-shaded-deps").file("src"))
-        val shadowJar = project(":sdk:trace-shaded-deps").tasks.named<Jar>("shadowJar")
-        from(zipTree(shadowJar.get().archiveFile))
-        dependsOn(shadowJar)
+    withType<AnimalSniffer>().configureEach {
```

Comment:
@jkwatson We should have had animalsniffer plugin on trace-shaded-deps, and it would have helped catch #3339 early. Now with the classes added to the main sourceset, the plugin in this configuration is taking care of it though. Give trace-shaded-deps isn't published, and it's really the job of anything "using it" to add constraints like animalsniffer, I guess this matches reality a bit better.